### ISSUE TITLE
`no-untyped-def` false positive on method overload with `/`

### ISSUE THREAD

**Issue by DetachHead** [KotlinIsland/basedmypy#367]
### Gist to reproduce
```python
class Foo:
    @overload
    def a(self, a: int, /) -> None: ...

    @overload
    def a(self, b: str, /) -> None: ...

    def a(self, c: int | str, /) -> None: # error: Function is missing a type annotation for one or more arguments  [no-untyped-def]
        ...
```

### Basedmypy version

1.4 / 1.5.0b1

# Workaround
Annotating the `self` resolves the error
